lib/rules/disallow-spaces-inside-function-calls.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Disallows space after opening and before closing function call parentheses.
+ *
+ * Types: `Boolean` or `Object`
+ *
+ * Values:
+ * - true
+ * - `Object`:
+ *      - "only": An array containing tokens to disallow.
+ *      - "onlySingleArgument": true
+ *      - "onlyMultipleArguments": true
+ *
+ * #### Example
+ *
+ * ```js
+ * "disallowSpacesInsideFunctionCalls": true
+ *
+ * // or
+ *
+ * "disallowSpacesInsideFunctionCalls": {
+ *     "allExcept": [ "'", "\'" ]
+ * }
+ * ```
+ *
+ * ##### Valid for mode `true`
+ *
+ * ```js
+ * doSomething(foo);
+ * ```
+ *
+ * ##### Valid for mode `{ only": [ "\'" ] }`
+ *
+ * ```js
+ * doSomething('foo');
+ * doSomething( bar );
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * doSomething( 'foo' );
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(value) {
+        var isObject = typeof value === 'object';
+
+        var error = this.getOptionName() + ' rule requires string value true or object';
+
+        var validFlags = [
+            'only',
+            'onlySingleArgument',
+            'onlyMultipleArguments'
+        ];
+
+        if (isObject) {
+            var hasValidFlags;
+
+            validFlags.forEach(function(flag) {
```

Comment:
Better to check keys to exist in validFlags. Like `assert(Object.keys(value).every(function (key) { return validFlags.indexOf(key) !== -1; }))`